README.md

```diff
@@ -1,2 +1,115 @@
+
+## Infrastructure setup
+
+The following steps assume the whole infrastructure will be set up.
+
+### Prepare CentOS system
+
+First, you'll need to prepare at least one CentOS system. This could be
+bare-metal, virtual machines or containers.
+
+### Execute Ansible playbooks
+
+There are Ansible playbooks that set up the Jenkins master node and two Jenkins
+slave nodes, one for Host OS builds and one for validating the code from GitHub
+pull requests. There could be multiple slave nodes set up for parallel builds.
+The different roles (master, builds slave and validation slave) could even be
+set up in the same system, as desired.
+
+Read the [Ansible instructions](ansible/README.md) for details on how to execute
+the playbooks.
+
+Note: The "master" playbook may fail due to Jenkins temporary unresponsiveness.
+If you see HTTP request errors, try running it again.
+
+### Manual Jenkins steps
+
+#### Seed job
+
+Open the server's webpage in your browser. It should show the Jenkins webpage,
+with a single job: "seed_job". Execute it, making sure the parameters are
+adequate for you infrastructure. With the default parameters, this will create
+all the other jobs configured to check out from the GitHub
+[open-power-host-os](https://github.com/open-power-host-os/) organization
+repositories.
+
+Note: If the "trigger" jobs are created before the GitHub credentials, those
+jobs will likely fail. To avoid that, execute the seed job with a glob that
+matches only the administrative jobs (e.g. "jenkins_jobs/create_*.groovy") and
+reexecute the seed job with the default "JOB_DESCRIPTORS_FILES" parameter as
+the last step.
+
+#### Slaves credentials
+
+Create the credentials necessary to access the slave nodes. This can be easily
+done by executing the job "create_ssh_credentials".
+
+#### GitHub credentials
+
+The GitHub Pull Request Builder plugin needs credentials with read access to be
+able to check out pull requests for building and validating and write access to
+update the pull request status, informing the developers of the job results.
+
+To create the credentials, you can use the "create_user_pass_credentials",
+passing either the user's password or it's API token as the "password"
+parameter.
+
+Refer to https://github.com/blog/1509-personal-api-tokens for information on
+how to create those tokens.
+
+Next, go to Jenkins home -> "Manage Jenkins" -> "Configure System" and look
+for "GitHub Pull Request Builder". Select the credentials you've created and
+set a meaningful description. The other fields do not need to be modified. You
+can use the "Test Credentials" button to make sure the permissions to your
+repository are correct. You'll need at least push and pull permissions.
+
+#### Create a slave node
+
+The default behaviour is for administrative job (creating jobs, nodes and
+credentials) to execute only on master node, and remaining jobs to execute only
+on slave nodes labeled "builds_slave_label" or "validation_slave_label". You'll
```

Comment:
Because "builds_slave" or "validation_slave" could be a slave name. Those were in fact the slave names in my test setup and are the defaults in the "create_slave_node" job.